utils/train_utils.py

```diff
@@ -101,16 +104,25 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
                         scaler.step(optimizer)
                         scaler.update()
                         optimizer.zero_grad()
-                        pbar.update(step//gradient_accumulation_steps)
+                        pbar.update()
+                        if step_scheduler:
+                            # Update the learning rate as needed
+                            lr_scheduler.step()
                 else:
                     # regular backpropagation when fp16 is not used
                     loss.backward()
                     if (step + 1) % gradient_accumulation_steps == 0 or step == len(train_dataloader) - 1:
                         optimizer.step()
                         optimizer.zero_grad()
-                        pbar.update(step//gradient_accumulation_steps)
+                        pbar.update()
+                        if step_scheduler:
```

Comment:
Is it relevant to update scheduler on different places? Would one not be sufficient and you just adapt the step length? Like we call update every step instead of every second but double step size?